internal/app/api/v1/executions.go

```diff
@@ -359,35 +360,26 @@ func (s TestkubeAPI) ExecutionLogsHandler() fiber.Handler {
 		ctx.SetBodyStreamWriter(fasthttp.StreamWriter(func(w *bufio.Writer) {
 			s.Log.Debug("starting stream writer")
 			w.Flush()
-			enc := json.NewEncoder(w)
 
-			// get logs from job executor pods
-			s.Log.Debug("getting logs")
-			var logs chan output.Output
-			var err error
-
-			logs, err = s.Executor.Logs(executionID)
-			s.Log.Debugw("waiting for jobs channel", "channelSize", len(logs))
+			execution, err := s.ExecutionResults.Get(ctx, executionID)
 			if err != nil {
-				output.PrintError(err)
-				s.Log.Errorw("getting logs error", "error", err)
+				output.PrintError(fmt.Errorf("could not get execution result for ID %s: %w", executionID, err))
+				s.Log.Errorw("getting execution error", "error", err)
 				w.Flush()
 				return
 			}
 
-			// loop through pods log lines - it's blocking channel
-			// and pass single log output as sse data chunk
-			for out := range logs {
-				s.Log.Debugw("got log", "out", out)
-				fmt.Fprintf(w, "data: ")
-				err = enc.Encode(out)
+			if isExecutionFinished(execution) {
```

Comment:
ExecutionResults has this method already

```
func (e *ExecutionResult) IsCompleted() bool {
```

Reply:
As I was testing it with the`test-api-local` tests, I saw that there is 2 more execution status `testkube.SUCCESS_Status` and `testkube.ERROR__Status`, so I checked for those as well in the `isExecutionFinished(...)` function. Tbh I was a bit confused as to why there are 2 different ways to describe the status, but wanted to make sure we catch both

Reply:
Holy moly! Good catch - we have file which is not used at all - some leftover after messing with moving status field to 

Please remove this file as it's not used and don't need to be handled anywhere. 

```
pkg/api/v1/testkube/model_status.go
```

And `execution.ExecutionResult.IsCompleted()` will be enough then

Reply:
Awesome, will do so!

Reply:
Do you know how we can check if there are more files like this?

Reply:
probably linter could partially do that - detect unused vars